hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -145,6 +146,11 @@ public void scan() {
           // hit a block with instant time greater than should be processed, stop processing further
           break;
         }
+        if (r.getBlockType() != CORRUPT_BLOCK && r.getBlockType() != COMMAND_BLOCK
+            && !timeline.containsOrBeforeTimelineStarts(r.getLogBlockHeader().get(INSTANT_TIME))) {
+          // hit an invalid block possibly from a failed write, move to the next one and skip processing this one
```

Comment:
this is not an invalid block right. this skipping an uncommitted block?

Reply:
Yes, that's why in the comments added "possibly from a failed write". Changed the invalid -> uncommitted